org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+
+/**
+ * BundleContext and Bundle utilities
+ */
+public class BundleUtils {
+
+	private static final String REFERENCE_PREFIX = "reference:";
+
+	/**
+	 * Load a list of bundle based on the provided file path locations.
+	 *
+	 * @param bundleLocations
+	 *            The list of bundle file path location
+	 * @throws CoreException
+	 *             throw the <code>CoreException</code> if failed to load any
+	 *             bundles.
+	 */
+	public static void loadBundles(List<String> bundleLocations) throws CoreException {

```

Comment:
`loadBundles(Collection<String> bundleLocations)` or `loadBundles(Set<String> bundleLocations)` if you really need to be more fine-grained